keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -638,24 +640,107 @@ always dispatched immediately.  Following is how the other requests
 are dispatched at a given apiserver.
 
 The concurrency limit of an apiserver is divided among the non-exempt
-priority levels in proportion to their assured concurrency shares.
-This produces the assured concurrency value (ACV) for each non-exempt
-priority level:
+priority levels, and higher ones can do a limited amount of borrowing
+from lower ones.
+
+Non-exempt priority levels are ordered in a total order for the
+purpose of borrowing currently-unused concurrency.  The ordering is
+based (first) on increasing value of a spec field whose name is
+`priority` and whose value is constrained to lie in the range 0
+through 10000 inclusive and (second) on increasing name of the
+PriorityLevelConfiguration object.  Priority levels that appear later
+in the order are considered to have lower priority.
+
+Borrowing is done on the basis of the current situation, with no
+consideration of opportunity cost and no pre-emption when the
+situation changes.
+
+An apiserver assigns three concurrency limits to each non-exempt
+priority level, with a constraint that means there are only two
+degrees of freedom.
+
+- The ***LendableConcurrencyLimit*** is the number of seats that are
+  statically (i.e., before any borrowing takes place) assigned to this
+  level and can be dynamically borrowed by higher levels.
+- The ***NonLendableConcurrencyLimit*** is the number of seats that
+  are statically assigned to this level and can _not_ be borrowed by
+  higher levels.
+- The ***NominalConcurrencyLimit*** is the number of seats statically
+  assigned to this level and is the sum of the
+  LendableConcurrencyLimit and the NonLendableConcurrencyLimit.
+
+Each non-exempt PriorityLevelConfiguration's spec has an
+`assuredConcurrencyShares`, which has existed since APF was introduced
+and may not be zero, and a `lendableConcurrencyShares` field, which is
+being added in the midst of the lifetime of the `v1beta2` version of
+the API and may be any value between zero and
+`assuredConcurrencyShares` inclusive (default is zero).  Each
+apiserver allocates NominalConcurrencyLimits in proportion to
+`assuredConcurrencyShares` and LendableConcurrncyLimit in
+corresponding propotion:
+
+```
+NominalConcurrencyLimit(i)  = ceil( SCL * assuredConcurrencyShares(i)  / sum_assured )
+LendableConcurrencyLimit(i) = ceil( SCL * lendableConcurrencyShares(i) / sum_assured )
```

Comment:
sum_assured? or sum_lendable?

Reply:
er, I guess I kind of expected both lines to use a sum of both kinds of shares.

Reply:
The lendable seats are a subset of the nominal seats, and the non-lendable seats are the complement.  If we had share values for those two categories then we would use the sum of both kinds of shares.  But I think that, for a smooth transition, we need to interpret the existing share value ("assuredConcurrencyShares") as the sum of the two categories and introduce a new share value for the lendable category.

Reply:
OK that's fine, either you added an explanatory comment or I didn't see it yesterday.